dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/InstanceVariables.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations.processor.data;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.base.Joiner;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+final class InstanceVariables {
+
+    private InstanceVariables() {}
+
+    static String joinCamelCase(String... segments) {
+        return Joiner.on("")
+                .join(IntStream.range(0, segments.length)
+                        .mapToObj(i -> {
+                            String segment = segments[i];
+                            CaseFormat caseFormat =
+                                    CaseFormats.estimate(segment).get();
```

Comment:
If estimation fails, should we return the input segment?

Could convert each segment to upper camel and convert the first letter of the result to lower. Doesn't make much difference.

Reply:
*I hope* that palantir-java-format saves us here? I guess this will be a pain to debug if someone actually hits this... I think I would probably prefer to validate this in the parsing stage, and throw exceptions here, as is?

Reply:
Heh, not everyone uses pjf yet, but I think most projects have some kind of style guidelines.
We'd need agreement between checkstyle, pjf, and this validation to stay somewhat in sync -- probably safe, but it would be painful if we discovered an edgecase.